### ISSUE TITLE
NoOpPasswordEncoder wrongly matches passwords

### ISSUE THREAD

**Issue by mmoayyed** [pac4j/pac4j#725]
If I have this:

```java
getPasswordEncoder().matches(credentials.getPassword(), null))
```

where `getPasswordEncoder` is a NoOp and `credentials.getPassword()` returns `password`. The result of this operation is `true`.

**Comment by leleuj** [comment]
You're right, unfortunately. It was introduced in 1.9.2: https://github.com/pac4j/pac4j/commit/a89c7c69c85de3af4f83e6cc6562c6847acee5b2#diff-0895a67466283b89a2aaba2a9990404aR20

The `MongoAuthenticator` and the `DbAuthenticator` are concerned.

I will fix it today, release a new 1.9.4, announce the issue and re-release implementations based on pac4j v1.9.2.

**Comment by leleuj** [comment]
Fixed in 1.9.x and master.
